integration/mediation-tests/tests-platform/tests-rabbitmq/src/test/java/org/wso2/carbon/esb/rabbitmq/utils/RabbitMQServerInstance.java

```
@@ -0,0 +1,99 @@
+        String host = "localhost";
+        try {
+            if (DOCKER_HOST.equalsIgnoreCase(
+                    TestConfigurationProvider.getAutomationContext().getConfigurationValue(RABBITMQ_HOST_XPATH))) {
+                String dockerHost = System.getenv("DOCKER_HOST");
+                if ((null != dockerHost) && (!StringUtils.isEmpty(dockerHost))) {
+                    URI uri;
+                    try {
+                        uri = new URI(dockerHost);
+                        host = uri.getHost();
+                    } catch (URISyntaxException e) {
+                        log.error("Error getting DOCKER_HOST", e);
+                    }
+                }
+            }
+
+        } catch (XPathExpressionException e) {
+            log.warn("Error reading the rabbitmq host in automation.xml. Proceed with default value " + host);
+        }
+        return host;
+    }
+
+    public static int getPort() {
+        //set default port
+        int port = 5672;
+        try {
+            port = Integer.parseInt(
+                    TestConfigurationProvider.getAutomationContext().getConfigurationValue(RABBITMQ_PORT_XPATH));
+        } catch (XPathExpressionException e) {
+            log.warn("Error reading the rabbitmq port in automation.xml. Proceed with default value " + port);
+        }
+        return port;
+    }
+
+    /**
+     * Initialize rabbitmq broker by declaring the the exchange and a routing queue that the producer will publishing
+     * to.
+     *
+     * @return the rabbitmq producer that was used to declaration.
+     */
+    public static RabbitMQProducerClient initRabbitMQBrokerWithProducer(String exchange, String routingKey) {
+        RabbitMQProducerClient sender = new RabbitMQProducerClient(getHost(), getPort(), USER_NAME, PASSWORD);
+        try {
+            sender.declareAndConnect(exchange, routingKey);
+        } catch (IOException e) {
+            Assert.fail("Could not connect to RabbitMQ broker");
```

No need to have the assert, we can throw the exception